src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -198,40 +201,6 @@ private static String convertStreamToString(InputStream is) throws IOException {
         }
     }
 
-    private static int getStatusCode(HttpURLConnection connection) throws IOException {
-        int statusCode = HttpURLConnection.HTTP_BAD_REQUEST;
-
-        try {
-            statusCode = connection.getResponseCode();
-        } catch (IOException ex) {
-
-            if (Build.VERSION.SDK_INT < 16) {
```

Comment:
I need you to help me understand what's changed? I may miss something, the only place I saw getStatusCode is called is in the send method, and it's called after getInputStream. The code comment here clearly shows that if the returned 401 challenge is empty, and when you call getResponseCode, you'll get exception when reading the code.